src/sage/combinat/affine_permutation.py

```diff
@@ -2014,13 +2014,13 @@ def __init__(self, cartan_type):
         """
         Parent.__init__(self, category=AffineWeylGroups())
         ct = CartanType(cartan_type)
-        self.k = ct.n
+        self.k = Integer(ct.n)
```

Comment:
Same question as earlier: from my reading of the code, `ct.n` will always be a Sage integer, so this cast should be unnecessary. Is there any reason you have `Integer(ct.n)` here instead of `ct.n`? I did some timing experiments and calling `Integer` does slow things down, and from the PR title this is supposed to be an optimization PR.

There are a few other locations in the code where you're calling `Integer` on something that I think is already going to be a Sage integer.